src/ngraph/op/embedding.cpp

```diff
@@ -0,0 +1,55 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/embedding.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+void op::Embedding::validate_and_infer_types()
+{
+    element::Type result_et = get_input_element_type(1);
+
+    const PartialShape& arg0_shape = get_input_partial_shape(0);
+    const PartialShape& arg1_shape = get_input_partial_shape(1);
+
+    PartialShape result_shape;
+
+    if (arg0_shape.rank().is_static() && arg1_shape.rank().is_static())
+    {
+        NODE_VALIDATION_ASSERT(this, static_cast<size_t>(arg1_shape.rank()) == 2)
```

Comment:
We should move this check up, outside the `if`, and make it something like this:

```
NODE_VALIDATION_ASSERT(this, arg1_shape.rank().is_dynamic() || static_cast<size_t>(arg1_shape.rank()) == 2)
    << "blah";
```

That way we'll fail if arg1's rank is bad, even if we don't know arg0's rank statically.